### PR TITLE
feat(cmd): pactus-shell support interactive shell

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

This feature make interactive shell Pactus-shell

[asciicast](https://asciinema.org/a/670989)

### CODE REVIEW

**Review by maxipaz — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -5,15 +5,19 @@ go 1.22.2
 require (
 	github.com/NathanBaulch/protoc-gen-cobra v1.2.1
 	github.com/beevik/ntp v1.4.3
+	github.com/c-bata/go-prompt v0.2.6
```
[the remark this answers is not in the document]
Comment (kehiy) [reply go.mod:8]: AFAIK, these libraries mostly work with stuff that rarely gets changed.

**Review by kehiy — APPROVED** [review]
LGTM

**Comment by b00f** [comment]
@Ja7ad is it possible we have both interactive and non-interactive?

**Comment by Ja7ad** [comment]
> @Ja7ad is it possible we have both interactive and non-interactive?
> 
> like: `pactus-shell -i` starts it in interactive format?

Sure but with command `shell`

**Comment by Ja7ad** [comment]
@b00f Currently support interactive and non-interactive shell.

You run as interactive with `pactus-shell shell`

```shell
pactus-shell is a command line tool for interacting with the Pactus blockchain using gRPC

Usage:
  shell [command]

Available Commands:
  blockchain  Blockchain service client
  clear       clear screen
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  network     Network service client
  shell       Start an interactive shell.
  transaction Transaction service client
  wallet      Wallet service client

Flags:
  -h, --help   help for shell

Use "shell [command] --help" for more information about a command.
```

File: util/shell/shell.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+			return nil
+		}
+		suggestions = parseSuggestions(out)
+		s.cache[key] = suggestions
+	}
+
+	return prompt.FilterHasPrefix(suggestions, d.GetWordBeforeCursor(), true)
+}
+
+func buildCompletionArgs(input string) ([]string, error) {
```
Comment (b00f) [inline util/shell/shell.go:138]: Did you test it for Windows and MacOS?

**Review by b00f — APPROVED** [review]

File: util/shell/shell.go (hunk 2 above)
Comment (Ja7ad) [reply util/shell/shell.go:138]: > Did you test it for Windows and MacOS?

Yes worked

File: util/shell/shell.go [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+// New creates a Cobra CLI command named "shell" which runs an interactive shell prompt for the root command.
+func New(root *cobra.Command, refresh func() *cobra.Command, opts ...prompt.Option) *cobra.Command {
+	sh := &lexer{
+		root:    root,
+		refresh: refresh,
+		cache:   make(map[string][]prompt.Suggest),
+	}
+
+	prefix := fmt.Sprintf("> %s ", root.Name())
+	opts = append(opts, prompt.OptionPrefix(prefix), prompt.OptionShowCompletionAtStart())
```
[the remark this answers is not in the document]
Comment (Ja7ad) [reply util/shell/shell.go:33]: > Let's get this as input option.

What's your mean?

**Pull request merged**